db/fetcher/encoded_doc.go

```diff
@@ -178,14 +183,32 @@ func convertToInt(propertyName string, untypedValue any) (int64, error) {
 
 // @todo: Implement Encoded Document type
 type encodedDocument struct {
+	mapping *core.DocumentMapping
+	doc     *core.Doc
+
 	Key        []byte
-	Properties map[client.FieldDescription]*encProperty
+	Properties []*encProperty
+
+	// tracking bitsets
+	// A value of 1 indicates a required field
+	// 0 means we we ignore the field
+	// we update the bitsets as we collect values
+	// by clearing the bit for the FieldID
+	filterSet *bitset.BitSet // filter fields
```

Comment:
suggestion: I'm not a fan of the fact that this property is only ever used from outside of this type - to me it strongly suggests that it should be moved off the `encodedDocument` type and held somewhere else, but perhaps I will warm to it as my brain-fog fades and the documentation increases :)

nitpick: Also feels a bit odd that this (and `doc`, and the `encProperty`s in `Properties`) is a pointer.

Reply:
Most of the properties of `encodedDocument` are directly accessed outside the type. Its really just a container type for holding state, and then its reponsible for combining state.

the doc `filterSet` could be on the `fetcher` directly, which would then have two sets, a "static" and a "dynamic" set, but as this is specific to the other fields within the `encodedDocument` type, it felt appropriate to keep here. Also means if I need to pass around the doc for any reason, the bitset goes with it.

re nitpick: Theres no good reason for the `encProperty` to be a pointer, but `BitSet` *must* be a pointer type as that's how its defined in its package. `New` method returns pointer, all methods are defined on pointer reciever, etc.

Reply:
> Most of the properties of encodedDocument are directly accessed outside the type. Its really just a container type for holding state, and then its reponsible for combining state.

Didnt realize you saw it that way - in that case I do think that this should have no funcs defined on it then as that means this type is used both as an object, and as a data-struct, which I really dislike. Change that is well out of scope regardless of what anyone thinks of it here though.  Convo resolved :)